clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -794,10 +794,6 @@ static void addPGOAndCoverageFlags(const ToolChain &TC, Compilation &C,
       Args.hasArg(options::OPT_coverage))
     FProfileDir = Args.getLastArg(options::OPT_fprofile_dir);
 
-  // TODO: Don't claim -c/-S to warn about -fsyntax-only -c/-S, -E -c/-S,
```

Comment:
Thanks for catching this. I left this in a07b135ce0c0111bd83450b5dc29ef0381cdbc39 to be cleaned up later but I forgot to remove this TODO.